### PR TITLE
Fix all compiler warnings

### PR DESCRIPTION
**Pull request by yyyeerbo**
Fix all warnings, to make compiler happy.

Signed-off-by: Yang Bo <bo@hyper.sh>

### CODE REVIEW

**Comment by yyyeerbo** [comment]
refer to issue https://github.com/kata-containers/kata-containers/issues/48

**Comment by yyyeerbo** [comment]
@jodhjodh-intel

File: rustjail/src/lib.rs [hunk 1]
```diff
@@ -3,6 +3,15 @@
...
 
+#![allow(unused_attributes)]
+#![allow(unused_imports)]
+#![allow(unused_variables)]
+#![allow(unused_mut)]
+#![allow(dead_code)]
+#![allow(deprecated)]
+#![allow(unused_must_use)]
+#![allow(non_upper_case_globals)]
+#![allow(unused_comparisons)]
```
Comment (jodh-intel) [inline rustjail/src/lib.rs]: Hi @yyyeerbo - these will of course silence the warnings, but can't we just remove the unused code? That means the codebase will be simpler and cleaner. We still have the history so we can extract the original version if required...?

Comment (yyyeerbo) [reply rustjail/src/lib.rs]: Removing unused code looks good to me. :)

**Comment by yyyeerbo** [comment]
several dead code will be needed for further development, so I leave it there for now .

**Pull request merged**